packages/ui/src/forum/hooks/useWatchlistedThreads.ts

```diff
@@ -17,16 +17,16 @@ interface UseMyThreads {
   pageCount?: number
 }
 
-export const useWatchlistedThreads = ({ page, threadsPerPage = 5 }: UseMyThreadsProps): UseMyThreads => {
+export const useWatchlistedThreads = ({ page = 1, threadsPerPage = 5 }: UseMyThreadsProps): UseMyThreads => {
   const [watchlist] = useLocalStorage<string[]>(FORUM_WATCHLIST)
 
   const status_json = {
     isTypeOf_not: 'ThreadStatusModerated',
   }
   const variables = {
     where: { id_in: watchlist ?? [], status_json },
-    limit: threadsPerPage,
-    offset: (page - 1) * threadsPerPage,
+    limit: threadsPerPage === -1 ? watchlist?.length || 5 : threadsPerPage,
+    offset: threadsPerPage === -1 ? 0 : (page - 1) * threadsPerPage,
```

Comment:
Currently this hook is used in only one place so the logic can be very simple:
1. `page` and `threadsPerPage` are not needed in the props.
2. no need to pass an `offset` and the limit is always `watchlist?.length`. It's fine if it's `undefined | 0` in this case the query should be skipped.
3. No need to return `pageCount`.
4. The `totalCount` is just `watchlist.length`, no need for `useGetForumThreadsCountQuery`

Finally I think a `skipQuery` property should be provided here so this hook can be called to know the `totalCount` without actually sending the query.